src/h5web/vis-packs/core/heatmap/MappedHeatmapVis.tsx

```diff
@@ -56,16 +62,13 @@ function MappedHeatmapVis(props: Props): ReactElement {
   const baseArray = useBaseArray(value, slicedDims);
   const dataArray = useMappedArray(baseArray, slicedMapping);
 
-  const domain = useMemo(() => {
-    return customDomain || getDomain(dataArray.data as number[], scaleType);
-  }, [customDomain, dataArray, scaleType]);
+  const dataDomain =
+    useDomain(dataArray.data as number[], scaleType) || DEFAULT_DOMAIN;
+  const visDomain = useVisDomain(dataDomain, customDomain);
```

Comment:
This `useVisDomain` hook will evolve to check whether the domain is "valid". Right now it's just a `useMemo` with the `??` logic that was previously in the JSX below. I might end up using it in the domain slider itself too.

Note that I can't get `dataDomain` from the config inside the hook without risking a stale render. I have to pass it as an argument.